src/bacnet/basic/object/csv.c

```diff
@@ -164,35 +185,100 @@ bool CharacterString_Value_Valid_Instance(uint32_t object_instance)
     unsigned index = 0; /* offset from instance lookup */
 
     index = CharacterString_Value_Instance_To_Index(object_instance);
-    if (index < MAX_CHARACTERSTRING_VALUES) {
-        return true;
+    for (index = 0; index < CSV_Max_Index; index++) {
+        if (object_instance == CSV_Descr[index].Instance) 
+        {
+            return true;
+        }
     }
 
     return false;
 }
 
+/**
+ * Initialize the charactestring inputs. Returns false if there are errors.
+ *
+ * @param pInit_data pointer to initialisation values
+ *
+ * @return true/false
+ */
+bool CharacterString_Value_Set(BACNET_OBJECT_LIST_INIT_T *pInit_data)
+{
+  unsigned i;
+
+  if (!pInit_data) {
+    return false;
+  }
+
+  if ((int) pInit_data->length > MAX_CHARACTERSTRING_VALUES) {
+    PRINTF("pInit_data->length = %d >= %d", (int) pInit_data->length, MAX_CHARACTERSTRING_VALUES);
+    return false;
+  }
+
+  for (i = 0; i < pInit_data->length; i++) {
+    if (pInit_data->Object_Init_Values[i].Object_Instance < BACNET_MAX_INSTANCE) {
+        CSV_Descr[i].Instance = pInit_data->Object_Init_Values[i].Object_Instance;
+    } else {
+      PRINTF("Object instance %u is too big", pInit_data->Object_Init_Values[i].Object_Instance);
+      return false;
+    }
+
+    if(!strcmp(CSV_Descr[i].Name, pInit_data->Object_Init_Values[i].Object_Name))
+    {
+        PRINTF("Fail to set Object name to \"%128s\"", pInit_data->Object_Init_Values[i].Object_Name);       
+    }
+    else
+    {   
+        strcpy(CSV_Descr[i].Name, pInit_data->Object_Init_Values[i].Object_Name);
+    }
+
+    if(!strcmp(CSV_Descr[i].Description, pInit_data->Object_Init_Values[i].Description))
+    {
+        PRINTF("Fail to set description to \"%128s\"", pInit_data->Object_Init_Values[i].Description);   
+    }
+    else
+    {
+        strcpy(CSV_Descr[i].Description, pInit_data->Object_Init_Values[i].Description);
+    }
+    
+  }
+
+  CSV_Max_Index = (int) pInit_data->length;
+
+  return true;
+}
+
 /**
  * For a given object instance-number, read the present-value.
  *
  * @param  object_instance - object-instance number of the object
- * @param  object_name - Pointer to the new BACnet string value,
+ * @param  Name - Pointer to the new BACnet string value,
  *                       taking the value.
  *
  * @return  true if values are within range and present-value
  *          is returned.
  */
+
 bool CharacterString_Value_Present_Value(
     uint32_t object_instance, BACNET_CHARACTER_STRING *object_name)
 {
     bool status = false;
     unsigned index = 0; /* offset from instance lookup */
-
+    
     index = CharacterString_Value_Instance_To_Index(object_instance);
-    if (object_name && (index < MAX_CHARACTERSTRING_VALUES)) {
+
+    if (index < CSV_Max_Index) 
```

Comment:
yeah lets use the:
 if (statement) {
} else {
}

style to remain consistent...